case-lib/lib.sh

```diff
@@ -797,6 +797,86 @@ initialize_audio_params()
     fi
 }
 
+set_sof_volume()
+{
+    local device=$1
+    local control_name=$2
+    local value=$3
+
+    case "$SOF_ALSA_TOOL" in
+        'alsa')
+            dlogc "amixer -c$device' -- cset 'name=$control_name' '$value'"
+            amixer -c"$device" cset name="$control_name" "$value" > /dev/null
+       ;;
+        'tinyalsa')
+           dlogc "tinymix -D'$device' set '$control_name' '$value'"
+            tinymix -D"$device" set "$control_name" "$value"  > /dev/null
+        ;;
+        *)
+           die "Unknown alsa tool $SOF_ALSA_TOOL"
+       ;;
+    esac
+}
+
+get_sof_controls()
+{
+    local sofcard=$1
+
+    case  "$SOF_ALSA_TOOL" in
+        'alsa')
+            amixer -c"$sofcard" controls
+       ;;
+        'tinyalsa')
+            tinymix --card "$sofcard" controls
+       ;;
+        *)
+            die "Unknown alsa tool $SOF_ALSA_TOOL"
+        ;;
+    esac
+}
+
+kill_play_record()
+{
+    case "$SOF_ALSA_TOOL" in
+        'alsa')
+           pkill -9 aplay
+        ;;
+        'tinyalsa')
+           pkill -9 tinyplay
+       ;;
+        *)
+           die "Unknown alsa tool $SOF_ALSA_TOOL"
+       ;;
+    esac
+}
+
+cleanup_and_fail()
+{
+    kill_play_record
+    die "$1"
+}
+
+check_alsa_tool_process()
+{
+    case "$SOF_ALSA_TOOL" in
+       "alsa")
+           # Check if the aplay process is running
+            if [[ ! $(pidof aplay) ]]; then
```

Comment:
Always prefer the exit status to the standard output when possible.
```suggestion
            pidof -q aplay ||
                  die ...
```
Avoiding negations is also more readable when possible. In this case it translates to plain English like "still_running or die".

Shorter _and_ more readable!